### PR TITLE
Fix incorrect resize behavior with first_block

### PR DESCRIPTION
**Pull request by pavelkumbrasev**
### Description 
Fixed `resize` behavior for case when new size lesser than old size. There is a bug in `concurrent_vector` that was reproduced when `resize `marked segment (that owned by `my_first_block`) as deleted ([code](https://github.com/oneapi-src/oneTBB/blob/f1ba69bde9e948594dde5d2d9dd094550bd81099/include/oneapi/tbb/detail/_segment_table.h#L189)) and segment allocation thread is stuck waiting this segment to be published by other thread that allocated first block ([code](https://github.com/oneapi-src/oneTBB/blob/f1ba69bde9e948594dde5d2d9dd094550bd81099/include/oneapi/tbb/concurrent_vector.h#L575)). 

Fixes # - 733
- [x] - git commit message contains appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/oneapi-src/oneTBB/blob/master/CONTRIBUTING.md#pull-requests) for details)_

### Type of change

- [x] bug fix - _change which fixes an issue_
- [ ] new feature - _change which adds functionality_
- [x] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [ ] documentation - _documentation update_

### Tests

- [x] added - _required for new features and for some bug fixes_
- [ ] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
@kboyarinov @anton-potapov

### CODE REVIEW

File: include/oneapi/tbb/concurrent_vector.h [hunk 1]
```diff
@@ -644,6 +644,22 @@ class concurrent_vector
         }
     }
 
+    segment_type nullify_segment( segment_table_type table, size_type segment_index ) {
+        segment_type target_segment{nullptr};
+        if (segment_index >= this->my_first_block) {
+            target_segment = table[segment_index].exchange(nullptr);
```
Comment (kboyarinov) [inline include/oneapi/tbb/concurrent_vector.h:650]: I would add relaxed here as well, up to you

File: test/tbb/test_concurrent_vector.cpp [hunk 2]
```diff
@@ -710,3 +710,29 @@ TEST_CASE("container_range concept for concurrent_vector ranges") {
...
+    while (v.size() < 737) {
+        v.emplace_back();
+    }
+
+    v.resize(1);
+    while (v.size() < 40) {
+        v.emplace_back();
+    }
+
+    v.resize(2222);
```
Comment (kboyarinov) [inline test/tbb/test_concurrent_vector.cpp:734]: I would add some assertions in this test, e.g. that vector size is correct

File: test/tbb/test_concurrent_vector.cpp [hunk 3]
```diff
@@ -710,3 +710,29 @@ TEST_CASE("container_range concept for concurrent_vector ranges") {
     static_assert(test_concepts::container_range<typename tbb::concurrent_vector<int>::const_range_type>);
 }
 #endif // __TBB_CPP20_CONCEPTS_PRESENT
+
+//! Test concurrent and unsafe operations
+//! \brief \ref regression
+TEST_CASE("testing resize on sequantual mode") {
```
Comment (kboyarinov) [inline test/tbb/test_concurrent_vector.cpp:716]: Please add the comment describing the problem we are solving

File: include/oneapi/tbb/concurrent_vector.h [hunk 4]
```diff
@@ -644,6 +644,23 @@ class concurrent_vector
         }
     }
 
+    segment_type nullify_segment( segment_table_type table, size_type segment_index ) {
+        segment_type target_segment{nullptr};
+        if (segment_index >= this->my_first_block) {
+            target_segment = table[segment_index].load(std::memory_order_relaxed);
+            table[segment_index].store(nullptr, std::memory_order_relaxed);
+        } else {
+            target_segment = table[segment_index].load(std::memory_order_relaxed);
```
Comment (pavelkumbrasev) [inline include/oneapi/tbb/concurrent_vector.h:653]:
```suggestion
        segment_type target_segment = table[segment_index].load(std::memory_order_relaxed);
        if (segment_index >= this->my_first_block) {
            table[segment_index].store(nullptr, std::memory_order_relaxed);
        } else {
```

File: include/oneapi/tbb/concurrent_vector.h [hunk 5]
```diff
@@ -644,6 +644,23 @@ class concurrent_vector
         }
     }
 
+    segment_type nullify_segment( segment_table_type table, size_type segment_index ) {
+        segment_type target_segment{nullptr};
```
Comment (kboyarinov) [inline include/oneapi/tbb/concurrent_vector.h]: Why does not make it as you suggested in the resolved conversation:

```suggestion
        segment_type target_segment = table[segment_index].load(relaxed);
        table[segment_index].store(nullptr, relaxed);
        if (segment_index == 0) {
            for (size_type i = 1; i < my_first_block; ++i) {
                table[i].store(nullptr, relaxed);
        }
        return target_segment;
```

**Review by kboyarinov — APPROVED** [review]

**Pull request merged**